proposals/0000-config-finalization.md

```diff
@@ -0,0 +1,160 @@
+- Start Date: 2022-03-22
+- Reference Issues: [[Roadmap] Astro v1.0](https://github.com/withastro/rfcs/discussions/1)
+- Implementation PR: <!-- leave empty -->
+
+# Summary
+
+Astro's configuration file is something we'd like to clean up prior to a v1.0 Beta release. Currently, it is not evident which specific options should go in which area, leading to user confusion and ocassional duplication.
+
+This RFC proposes a clean-up of the existing Astro config file format for consistency and clarity.
+
+# Example
+
+When this RFC is accepted, a config file for Astro might look something like this:
+
+```js
+import { defineConfig } from 'astro';
+import preact from '@astrojs/preact';
+
+export default defineConfig({
+    log: 'info',
+
+    rootDir: '.',
+    srcDir: './src',
+    publicDir: './public',
+    outDir: './dist',
+
+    site: 'https://example.com',
+    trailingSlash: 'always',
+    basePath: '/',
+
+    build: {
+        format: 'file',
+    },
+
+    server: {
+        port: 3000
+    },
+
+    dev: {
+        host: '0.0.0.0',
+    },
+
+    preview: {
+        host: false,
+        port: 8080
+    },
+
+    markdown: {
+        drafts: true,
+        syntaxHighlight: 'shiki',
+    },
+
+    sitemap: {
+        canonicalURL: 'https://example.com',
+        filter: (page) => page !== 'http://example.com/secret-page')
+    },
+
+    integrations: [
+        preact()
+    ],
+
+    vite: {}
+})
+```
+
+# Detailed design
+
+## Top-Level Options
+
+- Add `log`. Previously only supported through CLI flag.
```

Comment:
nit: Thoughts on `logLevel` to match Vite? `log` does sound a bit like a verb or a true/false boolean.

Reply:
This would be the only `camelCase` option, but I'm not opposed